DOC-5790: Ledger recovery section new structure (#1831)
corda/corda-docs-portal: content/en/platform/corda/4.11/enterprise/node/collaborative-recovery/ledger-recovery/privacy.md

---
date: '2023-11-08'
menu:
  corda-enterprise-4-11:
    parent: corda-enterprise-4-11-ledger-recovery
tags:
- ledger recovery privacy
title: Privacy
weight: 300
---

# Privacy

> The ledger recovery `DistributionList` is now encrypted using AES keys stored in the node's database.

Author: Updated

---
date: '2023-11-08'
menu:
  corda-enterprise-4-11:
    parent: corda-enterprise-4-11-ledger-recovery
tags:
- ledger recovery privacy
title: Privacy
weight: 300
---

# Privacy

The ledger recovery `DistributionList` is encrypted using AES keys stored in the node's database.

> Upon startup a node creates 10 random AES keys and stores them in the `node_aes_encryption_keys` table, if there are no keys already present.

Reviewer: ```suggestion
Upon startup, a node creates ten random AES keys and stores them in the `node_aes_encryption_keys` table, if there are no keys already present.
```

The keys themselves are obfuscated, by wrapping them with a deterministic AES key derived from the key's ID and the node's X.500 name.
{{< note >}}

> This is done purely to reduce the impact of an accidental data dump of the keys, and is not meant to be secure.

Reviewer: ```suggestion
This obfuscatation only reduces the impact of an accidental data dump of the keys, and is not meant to be secure.
```

{{< /note >}}

> The `senderRecordedTimestamp` has been moved to a separate header object, and is treated as the authenticated additional

Reviewer: Reads like a Release Note
```suggestion
`senderRecordedTimestamp` is in a separate header object, and is treated as the authenticated additional
```

> data in the AES-GCM encryption. This allows it to be public, which is necessary as the receiver node needs to read it

Reviewer: ```suggestion
data in the AES-GCM encryption. This allows it to be public, which is necessary to allow the receiver node to read it
```
